diff --git a/include/asterisk/extconf.h b/include/asterisk/extconf.h
index 1a04b01..4eaebb3 100644
--- a/include/asterisk/extconf.h
+++ b/include/asterisk/extconf.h
@@ -220,7 +220,7 @@ enum ext_match_t {
 	E_SPAWN =	0x12,	/* want to spawn an extension. Requires exact match */
 	E_FINDLABEL =	0x22	/* returns the priority for a given label. Requires exact match */
 };
-#define AST_PBX_MAX_STACK  128
+#define AST_PBX_MAX_STACK  512
 
 /* request and result for pbx_find_extension */
 struct pbx_find_info {
diff --git a/include/asterisk/pbx.h b/include/asterisk/pbx.h
index 770a1a9..8659eca 100644
--- a/include/asterisk/pbx.h
+++ b/include/asterisk/pbx.h
@@ -1548,7 +1548,7 @@ enum ext_match_t {
 #define STATUS_NO_PRIORITY	3
 #define STATUS_NO_LABEL		4
 #define STATUS_SUCCESS		5
-#define AST_PBX_MAX_STACK  128
+#define AST_PBX_MAX_STACK  512
 
 /* request and result for pbx_find_extension */
 struct pbx_find_info {
